ci_tools/compare_coverage_to_diff.py

```diff
@@ -0,0 +1,161 @@
+""" Functions for comparing coverage output and git diff output
+"""
+import os
+import sys
+import defusedxml.ElementTree as ET
+
+def get_untested_lines(coverage_filename):
+    """
+    Parse a coverage xml file and return a dictionary containing the files and lines
+    which are untested
+
+    Parameters
+    ----------
+    coverage_filename : str
+        The name of the xml file containing the coverage information
+
+    Result
+    ------
+    coverage : dict
+            A dictionary whose keys are the files in pyccel
+            and whose values are lists containing the line numbers
+            where coverage is lacking in that file
+    """
+    tree = ET.parse(coverage_filename)
+    root = tree.getroot()
+
+    content_lines = {}
+    changes = {}
+
+    for f in root.findall('.//class'):
+        filename = f.attrib['filename']
+        lines = f.findall('lines')[0].findall('line')
+        all_lines = [int(l.attrib['number']) for l in lines]
+        untested_lines = [int(l.attrib['number']) for l in lines if l.attrib['hits'] == "0"]
+        changes[os.path.join('pyccel',filename)] = untested_lines
+        content_lines[os.path.join('pyccel',filename)] = all_lines
+
+    return changes, content_lines
+
+def compare_coverage_to_diff(coverage, diff):
+    """
+    Compare dictionaries containing coverage information and git
+    diff information to find untested lines which have been added
+    to the code base
+
+    Parameters
+    ----------
+    coverage : dict
+            A dictionary whose keys are the files in pyccel
+            and whose values are lists containing the line numbers
+            where coverage is lacking in that file
+    diff     : dict
+            A dictionary whose keys are files which have been
+            changed in this branch and whose values are a dictionary.
+            The dictionary must contain a key 'addition' whose value
+            is a list containing the line numbers of lines which have
+            been changed/added
+
+    Result
+    ------
```

Comment:
I think the correct section title is `Returns`

Reply:
Fixed 44267e7a